### PR TITLE
Random maze

### PR DESCRIPTION
**Pull request by donno2048**
I replaced the old maze with a new one which is completely random (but still a valid one)

### CODE REVIEW

File: snakewm/apps/games/maze3D/maze3d.py [hunk 1]
```diff
@@ -1,13 +1,64 @@
...
+		if key in self.tree.keys() and element.parent != element:element.parent  = self.find(element.parent.key)
+		if key in self.tree.keys():return element.parent
+	def union(self, element_a, element_b):
+		root_a = self.find(element_a.key);root_b = self.find(element_b.key)
+		if root_a != root_b and root_a.rank < root_b.rank:root_a.parent = root_b
+		elif root_a != root_b and root_a.rank > root_b.rank:root_b.parent = root_a
+		elif root_a != root_b:root_b.parent = root_a;root_a.rank+=1
+
+class RandomMaze:
+	def __init__(self, width):self.width = width;self.height = width;self.seed = random.random()*10000;self.grid = [[(width*row + col) for row in range(0,width)] for col in range(0, width)];self.portals = {};self.kruskalize()
```
Comment (deeagle001) [inline snakewm/apps/games/maze3D/maze3d.py:32]: This kind of line raises the question in me that what style guide a Snakeware source shall follow?
Do you use the Google Python Style Guide or enforce PEP8?

Comment (donno2048) [reply snakewm/apps/games/maze3D/maze3d.py:32]: Kinda bad decisions, but I do whatever I want

Comment (joshiemoore) [reply snakewm/apps/games/maze3D/maze3d.py:32]: We use Black for formatting, the style gets "auto-blacked" once it's merged into master. This looks cool though, I'll check it out soon!

Comment (deeagle001) [reply snakewm/apps/games/maze3D/maze3d.py:32]: Oh. Thanks! Does not even know about Black till now. Today I learned something new. :)
I just questioned the format, as it is pretty hard to review long compact lines with no separators as they are not easy for the human eyes.

Comment (donno2048) [reply snakewm/apps/games/maze3D/maze3d.py:32]: Oops, sorry...

**Pull request merged**